teps/0007-conditions-beta.md

```diff
@@ -348,6 +353,106 @@ spec:
       - echo-file-does-not-exist
 ```
 
+### Status
+
+Add `Skipped Tasks` section to the `PipelineRunStatus` that contains a map with `PipelineTaskName` as key and resolved `WhenExpressions` as value. Thus, users can know why a particular `Task` was skipped. In addition, `TaskRuns` from guarded `Tasks` that execute because their `WhenExpressions` evaluate to true would have the resolved `WhenExpressions` included in the `TaskRun` status. 
+
+In this example, the `WhenExpressions` in `skip-this-task` evaluate to False while the `WhenExpressions` in `run-this-task` evaluate to True:   
+
+```yaml
+Status:
+  Completion Time:  2020-08-27T15:07:34Z
+  Conditions:
+    Last Transition Time:  2020-08-27T15:07:34Z
+    Message:               Tasks Completed: 1 (Failed: 0, Cancelled 0), Skipped: 1
+    Reason:                Completed
+    Status:                True
+    Type:                  Succeeded
+  Pipeline Spec:
+    Tasks:
+      Name:  skip-this-task
+      Task Spec:
+        Metadata:
+        Steps:
+          Image:  ubuntu
+          Name:   echo
+          Resources:
+          Script:  echo "Good Night!"
+      When:
+        Input:     foo
```

Comment:
A key difference between `input` here and `input` under the list of `Skipped Tasks` is unresloved `param` i.e. 

```
Input: $(params.param)
```

After resolving it:

```
Input: foo
```

Reply:
yes yes, added a variable to clarify that they are unresolved in the pipelinespec and resolved in the when expressions status